tests/networkpolicy_test.go

```diff
@@ -2,41 +2,58 @@ package tests_test
 
 import (
 	"fmt"
+	"time"
 
-	expect "github.com/google/goexpect"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	v1network "k8s.io/api/networking/v1"
 	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
 	v1 "kubevirt.io/client-go/api/v1"
 	"kubevirt.io/client-go/kubecli"
 	"kubevirt.io/kubevirt/tests"
 	cd "kubevirt.io/kubevirt/tests/containerdisk"
 )
 
-func assertPingSucceed(ip string, vmi *v1.VirtualMachineInstance) {
-	expecter, err := tests.LoggedInCirrosExpecter(vmi)
-	Expect(err).ToNot(HaveOccurred())
-	defer expecter.Close()
+func pingEventually(fromVmi *v1.VirtualMachineInstance, toIp string) AsyncAssertion {
+	return Eventually(func() error {
+		By(fmt.Sprintf("Pinging from VMI %s/%s to Ip %s", fromVmi.Namespace, fromVmi.Name, toIp))
+		return tests.PingFromVMConsole(fromVmi, toIp)
+	}, 15*time.Second, time.Second)
+}
 
-	err = tests.CheckForTextExpecter(vmi, []expect.Batcher{
-		&expect.BSnd{S: fmt.Sprintf("ping -w 3 %s \n", ip)},
-		&expect.BExp{R: "0% packet loss"},
-	}, 60)
-	Expect(err).ToNot(HaveOccurred())
+func assertPingSucceedBetweenVMs(vmisrc, vmidst *v1.VirtualMachineInstance) {
+	ExpectWithOffset(1, vmidst.Status.Interfaces[0].IPs).NotTo(BeEmpty())
+	for _, ip := range vmidst.Status.Interfaces[0].IPs {
+		pingEventually(vmisrc, ip).Should(Succeed())
+	}
 }
 
-func assertPingFail(ip string, vmi *v1.VirtualMachineInstance) {
-	expecter, err := tests.LoggedInCirrosExpecter(vmi)
-	Expect(err).ToNot(HaveOccurred())
-	defer expecter.Close()
+func assertPingFailBetweenVMs(vmisrc, vmidst *v1.VirtualMachineInstance) {
+	ExpectWithOffset(1, vmidst.Status.Interfaces[0].IPs).NotTo(BeEmpty())
+	for _, ip := range vmidst.Status.Interfaces[0].IPs {
+		pingEventually(vmisrc, ip).ShouldNot(Succeed())
+	}
+}
 
-	err = tests.CheckForTextExpecter(vmi, []expect.Batcher{
-		&expect.BSnd{S: fmt.Sprintf("ping -w 3 %s \n", ip)},
-		&expect.BExp{R: "100% packet loss"},
-	}, 60)
+func setupVMIWithEphemeralDiskAndUserdata(virtClient kubecli.KubevirtClient, namespace string, labels map[string]string) *v1.VirtualMachineInstance {
+	var err error
+	vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), "#!/bin/bash\necho 'hello'\n")
+	vmi.Namespace = namespace
+	vmi.Labels = labels
+	vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Create(vmi)
 	Expect(err).ToNot(HaveOccurred())
+
+	return tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
+}
+
+func waitForNetworkPolicyDeletion(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, policyName string) {
+	Eventually(func() bool {
+		_, err := virtClient.NetworkingV1().NetworkPolicies(vmi.Namespace).Get(policyName, v13.GetOptions{})
+		return apierrors.IsNotFound(err)
+	}, 10*time.Second, time.Second).Should(BeTrue())
```

Comment:
Let's make it more verbose to make easier to debug, my previous reference was not totally right the more verbose version of this would be https://github.com/nmstate/kubernetes-nmstate/blob/master/test/e2e/handler/utils.go#L186-L194

verbose means, printing the not matched error to know what `.Get` is returning and adding an description string at `.Should`

Reply:
Done thx